array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,37 @@
+/* exported zip */
+// .concat both arrays into an empty array;
```

Comment:
Your solution works, but it's fairly complex. FYI, a simpler solution would be to walk through both arrays at the same time, using something like:
```
for (var i = 0; i < first.length; i++) {
  newArray.push([first[i], second[i]]);
}
```
You would need to make sure you don't loop too many times if the two arrays have different lengths (you can use Math.min to get the smallest of the two).